### DISCUSSION TITLE
Where does the android app stores locations?

### DISCUSSION THREAD

**Discussion by ippocratis** [owntracks/android#1772] [Q&A]
I'm experimenting with  location history visulisation making use of the Owntracks .rec files from the recorder. 
I was thinking that maybe I could also try to make all the work locally on android if I could access the locations directly from the android app.

I have root so getting in to the app Dir is not a problem .

I just can't figure out how and where the locations are stored locally .

Thanks...

**Comment by jpmens** [comment]
Locations are stored locally in Android only for the duration until they've been successfully transmitted. There's no long-term storage there.

**Reply by ippocratis** [reply]
I see .
Makes sense I guess since owntracks is designed to work with a server.
Thanks for clarifying.

**Comment by jpmens** [comment]
Correct.

What you could do, if this is something you want and can, is to query the Recorder API for data which you then display. This would obviously mean having access to the Recorder (or backend store) and/or caching the data it provides. But this is way out of scope of what our apps want to do.

**Reply by ippocratis** [reply]
I'm running python scripts against the .rec files from the store at the moment.
Never thought to call recorder directly for the json data and use the script on  that.
Thanks for the tip
